batch_job/src/batch_task.c

```diff
@@ -133,4 +147,120 @@ void batch_task_set_info(struct batch_task *t, struct batch_job_info *info)
 	t->info->disk_allocation_exhausted = info->disk_allocation_exhausted;
 }
 
+struct batch_task_wrapper *batch_task_wrapper_create(void) {
+	return xxcalloc(1, sizeof(struct batch_task_wrapper));
+}
+
+void batch_task_wrapper_delete(struct batch_task_wrapper *w) {
+	if (!w) return;
+	list_free(w->pre);
+	list_delete(w->pre);
+	list_free(w->post);
+	list_delete(w->post);
+	list_free(w->cmd);
+	list_delete(w->cmd);
+	free(w);
+}
+
+void batch_task_wrapper_pre(struct batch_task_wrapper *w, const char *cmd) {
+	assert(w);
+	assert(cmd);
+	assert(!w->cmd);
```

Comment:
Are you requiring that `w->cmd` is not specified prior to the pre or post? Is there a purpose to asserting this here? Also in post at line 206.

Reply:
That's basically because of laziness. I have uses in mind for multiple pre commands followed a single argv. Rather than adding in structures to support arbitrary combinations of commands, I just implemented the former usage and put in those asserts so we could add the full functionality (if it's needed down the road) without breaking any existing code. I can add a comment explaining that this is a limited subset of functionality.